dip-0006.md

```diff
@@ -369,21 +369,30 @@ A final commitment is mined in the payload of a new special transaction (DIP2).
 
 With these new consensus rules, the incentives for miners are turned around. Instead of requiring an incentive to include it in a block, we remove the incentive to leave it out. Whatever the miner decides, the space required for the commitment is always used. If a miner still decides to be dishonest, other honest miners will include the special transaction in one of the next blocks.
 
 These consensus rules are only applied while the DKG is in the mining phase. After a block is mined that includes a final commitment, the consensus rules are not applied anymore (even if still in the mining phase). This ensures that not too much space is wasted.
 
 Every node, that receives a block, must verify that these consensus rules are followed. This means:
 
-1. If the DKG is not in the mining phase and the new block contains a (null or non-null) commitment, the block should be rejected.
-2. If no final commitment was mined for the current DKG yet and the new block does not include a (possibly null) commitment, the block should be rejected.
-3. If in the current mining phase a previous block already mined a non-null commitment and the new block contains another (null or non-null) commitment, the block should be rejected.
-4. If the block contains more than one commitment, the block should be rejected.
+1. If the "height" field of the special transaction payload does not match the height of the block in which it the payload is included, the block should be rejected. This ensures that all commitment special transactions have a unique hash, including null commitments.
+2. If the DKG is not in the mining phase and the new block contains a (null or non-null) commitment, the block should be rejected.
+3. If no final commitment was mined for the current DKG yet and the new block does not include a (possibly null) commitment, the block should be rejected.
+4. If in the current mining phase a previous block already mined a non-null commitment and the new block contains another (null or non-null) commitment, the block should be rejected.
+5. If the block contains more than one commitment, the block should be rejected.
 
 After a block with a valid (non-null) commitment has been accepted and processed, the quorum belonging to the commitment must be added to the active quorums set. From this point on, the quorum is able to perform threshold signing of arbitrary messages and all nodes in the network are able to verify the result (recovered signature) of such signing sessions.
 
+The format of the "quorum commitment" special TX payload with type=6 is:
+
+| Field | Type | Size | Description |
+|--|--|--|--|
+| height | int | 4 | The height of the block in which this commitment is included
```

Comment:
`height` should be of `uint32_t` type. Also, no `version`?

Reply:
changed height to uint32_t and added version